c7n/resources/appmesh.py

```diff
@@ -0,0 +1,280 @@
+        detail_spec = ('describe_mesh', 'meshName', 'meshName', None)
+
+        # This function gets called with the id's retrieved from the
+        # event and is expected to return the enriched resource
+        # definition.
+        #
+        # Explanation of logic:
+        #
+        # The default impl on the superclass calls the enum_spec
+        # function, however for many resource types this is useless
+        # since the enum function provides little specification and
+        # often everything the enum function provides is also on the
+        # detail_spec function.
+        #
+        # So where we're going to have to have a detail_spec and where
+        # the enum_spec is redundant then why call the enum spec
+        # function at all?  This is a small speed / cost saving as it
+        # reduces the number of AWS API calls made for each event.
+        # So, in the the case of the current resource we will simply
+        # skip the enum call entirely.
+        #
+        # TODO: Propose an improvement where we allow skipping of the
+        # enum_spec call self.source.get_resources(ids)
+
+    def get_resources(self, ids, cache=True, augment=True):
+        if not ids:
+            return []
+        if cache:
+            resources = self._get_cached_resources(ids)
+            if resources is not None:
+                return resources
+        try:
+            # default impl calls the enum function here - but we're
+            # not going to do that
+            # resources = self.source.get_resources(ids)
+
+            # do the augment by mocking up a fake resource that has
+            # the field name (param_key)
+            # specified by the describe_spec
+            resources = [{"meshName": m} for m in ids]
+            return self.augment(resources)
+        except ClientError as e:
+            self.log.warning("event ids not resolved: %s error:%s" % (ids, e))
+            return []
+
+
+@resources.register('appmesh-virtual-gateway')
+class AppmeshVirtualGateway(ChildResourceManager):
+    # interior class that defines the aws metadata for resource
+    # see c7n/query.py for documentation on fields.
+    class resource_type(TypeInfo):
+        # turn on support for cloundtrail for child resources
+        supports_trailevents = True
+
+        service = 'appmesh'
+        arn_type = "virtualGateway"
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appmesh-virtualgateway.html
+        # probably wrong as the arn is a different field
+        # OLD arn = "meshName"
+        # arn = "metadata.arn"
+
+        id = name = 'meshName'
+        date = 'createdAt'
+
+        # when we define a parent_spec then it uses the parent sNpec
+        # to provide the driving result set.  this is then iterated
+        # across and the enum_spec is called for each parent instance.
+        # appmesh-mesh - is ref to another resource above that

```

Comment:
again default behavior, its also why trail event support for child resources requires being explicit, ie default is off, if the event contains the parent id the source can directly fetch the specific parent and child.